pkg/server/commands/reverseexpand/reverse_expand.go

```diff
@@ -334,6 +336,9 @@ func (c *ReverseExpandQuery) reverseExpandTupleToUserset(
 
 	combinedTupleReader := storagewrappers.NewCombinedTupleReader(c.datastore, req.contextualTuples)
 
+	if ctx.Err() != nil {
```

Comment:
Checking context error between every line of code has got to be an anti-pattern. Instead, the check should be at the iterator level (as in https://github.com/openfga/openfga/pull/1055).

We should be avoiding dispatching more work if the context has been cancelled, but checking it between every function call is extreme.

Reply:
> Checking context error between every line of code has got to be an anti-pattern.

Ah yes i didn't like it either, i only put it before the expensive database calls. 

> Instead, the check should be at the iterator level (as in #1055).

This sounds more reasonable 😆

Reply:
I'm going to see if I can resolve the issue that @adriantam reported in #1055 and then we can rebase on that. The other changes are awesome.